### ISSUE TITLE
[YSQL] Second connection to template1 crashes

### ISSUE THREAD

**Issue by frozenspider** [yugabyte/yugabyte-db#13796]
Jira Link: [DB-3312](https://yugabyte.atlassian.net/browse/DB-3312)
### Description

When connecting to `template1` database after cluster startup, only the first connection would go through. All subsequent would crash:
```
$ ./ysqlsh -h localhost template1
ysqlsh (11.2-YB-2.15.3.0-b0)
Type "help" for help.

template1=#^D\q

$ ./ysqlsh -h localhost template1
ysqlsh: server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
```
Logs:
```
F0829 18:39:21.803912 95651200 pg_sys_table_prefetcher.cc:349] Sys table prefetching is enabled but requested table { database_oid: 1 object_oid: 1259 } was not registered. The list of tables ready for prefetching is: []
Fatal failure details written to /Users/fs/yugabyte-data/node-1/disk-1/yb-data/tserver/logs/yb-tserver.FATAL.details.2022-08-29T18_39_21.pid36047.txt
```
This is related to the fact that `global/pg_internal.init` file is not available for the first run so the full preload is executed, and this snippet in `relcache.c`:
```c
	/*
	 * During initdb also preload catalog caches (not just relation cache) as
	 * they will be used heavily.
	 */
	if (IsYugaByteEnabled() && YBIsPreparingTemplates())
		YBPreloadCatalogCaches();
```

**Comment by frozenspider** [comment]
Apparently, this only happens when the cluster is created using an older initdb snapshot w/o upgrade

**Comment by frozenspider** [comment]
Resolved in 6cffdb5
